platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java

```diff
@@ -169,9 +199,99 @@ private ImportMigrationConfigurationAdminEntry updateNecessaryConfigurationAdmin
       entry.setPid("ddf.security.http.impl.HttpSessionFactory");
     }
+    // this pid was updated in version 2.17.0
+    // see https://github.com/codice/ddf/pull/5105
+    if (entry.getPid().equals("ddf.security.sts.guestclaims")) {
+      entry.setPid("ddf.security.guest.realm");
+    }
+    // this pid was updated in version 2.17.0
+    // see https://github.com/codice/ddf/pull/5105
+    if (entry.getPid().equals(WEB_CONTEXT_POLICY_MANAGER_PID)) {
+      entry = updateWebContextPolicyManagerConfiguration(entry);
+    }
+    return entry;
+  }
+
+  private ImportMigrationConfigurationAdminEntry updateWebContextPolicyManagerConfiguration(
+      ImportMigrationConfigurationAdminEntry entry) {
+    Map<String, Object> defaults = getDefaultProperties(WEB_CONTEXT_POLICY_MANAGER_PID);
+    Dictionary<String, Object> props = entry.getProperties();
+    props.remove("realms");
+
+    String[] whiteListContexts = (String[]) props.get(WHITE_LISTED_CONTEXTS);
+    String[] whiteListContextDefaults = (String[]) defaults.get(WHITE_LISTED_CONTEXTS);
+
+    Set<String> whiteListContextsSet =
+        Stream.of(whiteListContexts, whiteListContextDefaults)
+            .flatMap(Stream::of)
+            .collect(Collectors.toSet());
+
+    props.put(WHITE_LISTED_CONTEXTS, String.join(",", whiteListContextsSet));
+
+    String[] authenticationTypes = (String[]) props.get(AUTHENTICATION_TYPES);
+    List<String> authenticationTypesList = new ArrayList<>(Arrays.asList(authenticationTypes));
+
+    for (int i = 0; i < authenticationTypesList.size(); i++) {
+      String val = authenticationTypesList.get(i);
+      String[] parts = val.split("=");
+
+      if (parts.length == 2) {
+        String[] auths = parts[1].split("\\|");
+        List<String> list =
+            Arrays.stream(auths)
+                .filter(a -> !StringUtils.containsIgnoreCase(a, "guest"))
+                .map(a -> a.replaceAll("(?i)idp", "SAML"))
+                .collect(Collectors.toList());
+        authenticationTypesList.set(i, String.format("%s=%s", parts[0], String.join("|", list)));
+      }
+    }
+
+    props.put(AUTHENTICATION_TYPES, String.join(",", authenticationTypesList));
+
+    if (props.get(GUEST_ACCESS) == null) {
+      props.put(GUEST_ACCESS, "true");
+    }
+    if (props.get(SESSION_ACCESS) == null) {
+      props.put(SESSION_ACCESS, "true");
+    }
+
+    entry.setProperties(props);
     return entry;
   }
 
+  Map<String, Object> getDefaultProperties(String pid) {
+    Optional<Service> defaultMetatypeValues =
+        configAdminMBean
+            .listServices()
+            .stream()
+            .filter(service -> service.getId() != null && service.getId().equals(pid))
+            .findFirst();
+
+    List<MetatypeAttribute> metatypes = new ArrayList<>();
+    if (defaultMetatypeValues.isPresent()) {
+      metatypes = defaultMetatypeValues.get().getAttributeDefinitions();
+    }
```

Comment:
❗️ We really should have an else block here and throw an exception, if so. None of the `PolicyManager` conversion logic will work if we can't get the default values (I think as it is now, it'll throw an NPE).

Reply:
Do we want the throw to be like this? 
https://github.com/codice/ddf/blob/3efc289a995afbddf1170c716233a34ee82eaab8/platform/admin/configurator/admin-configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/ServiceOperation.java#L102
Also, how would we want to bubble/catch this exception?

Reply:
I think we should probably do
- something like this, but records a MigrationWarning instead of an exception and log that the PolicyManager config is not getting upgraded
https://github.com/codice/ddf/blob/3efc289a995afbddf1170c716233a34ee82eaab8/platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java#L142-L148
- then return `null`
- then check for `null` in the `updateWebContextPolicyManagerConfiguration` method. If it's `null`, then return the `entry` as-is.